### ISSUE TITLE
[4.3 beta 1] Godot Wayland produces corrupt config if language does not use '.' as decimal separator

### ISSUE THREAD

**Issue by DasCapschen** [godotengine/godot#92594]
### Tested versions

- v4.3.beta1.official [a4f2ea91a]

### System information

Godot v4.3.beta1 - openSUSE Tumbleweed 20240523 - Wayland - Vulkan (Forward+) - dedicated AMD Radeon RX 580 Series (RADV POLARIS10) () - Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz (8 Threads)

### Issue description

Godot writes invalid `editor_settings-4.3.tres` and invalid `project.godot` files, when the User's language does not use a period as a decimal separator.  
Floats get represented like `0,57` instead of `0.57`, causing issues like `Parse Error: Expected 4 arguments for constructor`.

This prevents Godot from starting until editor_settings file is fixed.  

Starting Godot with `LANG=C ./godot` fixes the problem. 

### Steps to reproduce

- run Godot 4.3 Beta 1 with PC language set to German
- change colors in Theme settings of the Editor
- close Godot
- start Godot
- crashes immediately, `Parse Error: Expected 4 arguments for constructor`

### Minimal reproduction project (MRP)

n.a.

**Comment by DasCapschen** [comment]
Addition: 
This might be related to Wayland. After starting the Editor for the first time, I immediately turned on Wayland and clicked "save & restart". The engine froze. I killed the process, restarted, and the config file was invalid.

**Comment by Calinou** [comment]
Can you reproduce this in 4.2.2? I've never heard of this issue before, as Godot does not use language-specific decimal separators when printing numbers in the first place.

**Comment by DasCapschen** [comment]
No, I've been using 4.2.2 (v4.2.2.stable.openSUSE [15073afe3] to be exact) the whole time and never had this issue.

**Comment by DasCapschen** [comment]
I just tried again, and it only happens with `prefer_wayland = true`.   

I'll attach a Screencast:

[Screencast_20240601_125532.webm](https://github.com/godotengine/godot/assets/3923406/a40548ff-90e3-4282-a0e7-783cddd1ddfe)

**Comment by DasCapschen** [comment]
Well, the [Man-Page](https://linux.die.net/man/3/sprintf) for `sprintf` says the decimal separator depends on `LC_NUMERIC`.  So setting `LC_CTYPE` should not change that.  But maybe it is worth giving that a try anyways:

- the [Unix Env Var Spec](https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html) says that it will use `LANG` in absence of `LC_ALL` and other `LC_*` variables.  I wonder if this is true for all the LC_* variables.  So perhaps if `LC_NUMERIC` is unset, it uses uses the next best thing (perhaps `LC_CTYPE`, if set). 


Other than that, I guess the only reason it happens with Wayland is that Wayland sets (or doesn't set) these Variables differently from X11.

**Comment by DasCapschen** [comment]
Well, I just inspected `/proc/<pid>/environ` to find differences between running X11 and Wayland, and... there is absolutely no difference... if I set `LANG=C`, the only difference is that `LANG=C` instead of `LANG=de_DE.UTF-8` ...  
Interestingly, apart from `LANG`, no other `LC_*` variables are set at all!

Setting `LC_NUMERIC=C` does solve the problem!  
Setting `LC_CTYPE=C` does **not** solve the problem.

**Comment by DasCapschen** [comment]
It definitely happens *inside* `snprintf`
Here's an `ltrace` of Godot without `prefer_wayland`:
```
[pid 21555] Godot_v4.3-beta1_linux.x86_64->snprintf("0.56", 256, "%lg", 0,560000) = 4
```

And now with `prefer_wayland`...
```
[pid 21740] Godot_v4.3-beta1_linux.x86_64->snprintf("0,56", 256, "%lg", 0,560000) = 4
```

As you can see, the output with `prefer_wayland` takes locale into account, while without it, it does not 🤔   
So... is something weird going on inside libc?!

**Comment by DasCapschen** [comment]
Hi, sorry for the late response, haven't used Godot in a while. I just tried it today with Version `v4.4.1.stable.mono.official [49a5bc7b6]` using my own steps to reproduce from above, and the problem did not occur anymore :)

**Comment by DasCapschen** [comment]
Yeah, I guess it can be closed :)
